### PR TITLE
Material class

### PR DESCRIPTION
**Pull request by SteSeg**
This PR implement the Material class for the "material database" project.

Main class looks like this:
```
@dataclass
class Material:
    """Material object that has all the material properties wanted
    """
    name: str
    composition: str
    density: Union[float, Callable]
    thermal_conductivity: Union[float, Callable] = None
    thermal_expansion_coefficient: Union[float, Callable] = None
    melting_temperature: float = None
    yield_strength: float = None
```
It is possible to create a Material by defining at least the `name`, the `composition` and the `density` 

it is possible to save the material into an .pkl file that either exists or gets generated with:

`Material.save_to(filename)`

### CODE REVIEW

File: fusion_toolbox/material_constructor.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        """
+
+        try:
+            dc = DataContainer(load_fn=filename)
+        except FileNotFoundError:
+            dc = DataContainer()
+            dc.trials = []
+
+        dc.trials.append(self)
+        dc.save('material_database.pkl')
```
Comment (RemDelaporteMathurin) [inline fusion_toolbox/material_constructor.py:36]: Should the path of the saved file be exposed to the users?

**Review by RemDelaporteMathurin — COMMENTED** [review]
Looks good to me, can you write a test for the `save_to` method?

**Comment by RemDelaporteMathurin** [comment]
@SteSeg the CI is failing because `dataserve` is not installed.
You'll need to add it to the requirements.txt file

**Review by eepeterson — CHANGES_REQUESTED** [review]
nice work! maybe rename the file to just `material.py`

**Pull request merged**